### ISSUE TITLE
[HOLD for payment 2023-05-22] [$4000] Not showing any data on search page when press Cmd+K shortcut while loading after login

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14490]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Update network connection to Slow 3G (to reproduce easily)
2. Login with any account
3. While full screen loading, press Cmd+K shortcut
4. Search page opens

This also applies to other pages like Profile, New chat, Send money, Workspaces, not only search

## Expected Result:
* When the user launches the app and reports data is still loading, they can open the search page with `CMD+K` as usual
* The search page opens, and displays a skeleton UI (just as the LHN would when it doesn’t yet have reports data)
* While the search page is loading, the options in the header are not disabled. So the user can:
  * Enter a search query (still, just a skeleton UI is shown)
  * Press < to close the RHP
  * Tap outside or swipe to close the RHP
* If the reports load while the search page is displaying a skeleton UI, then any reports matching the search query are displayed. If there are no results matching the search query, then the skeleton UI is replaced by the standard `No results found` UI

## Actual Result:
shows nothing on Search page

## Workaround:
Close and re-open Search page

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] MacOS / Chrome / Safari

**Version Number:** 1.2.58-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/214165578-fc7d368e-1cea-4f08-8aab-c06ac416de9f.mov

https://user-images.githubusercontent.com/43996225/214165723-4608bbb5-0922-4a6f-a739-fc4a7c5c5eb1.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @aimane-chnaif 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1674469304602449

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013d319d96dd608a1f</li>
        <li>Upwork Job ID: 1620163656058773504</li>
        <li>Last Price Increase: 2023-03-31</li>
    </ul>
</details>

**Comment by mallenexpensify** [comment]
For the workaround, you can also start typing then delete text and search will populate.  
I'm unsure this is a bug because there's will always(?) be a time between clicking `Sign in`  in and being able to take any action.  ie. there'll always be a loading spinner, even if it's only a fraction of a second - are all app functions that don't work in that that fraction of a second bugs? 

Coincidentally... we're working on high-level testing guides for QA, I wonder if we should include "After clicking "`Sign in", app should load within x seconds (cuz... in both examples above it took multiple seconds to load)

I'm going to keep this assigned for now and unlock it for feedback

**Comment by mallenexpensify** [comment]
Thanks for pointing me to the Slack thread @aimane-chnaif , I forgot to check it :ohnothing:
@amyevans , I'm unsure if this can be External.

**Comment by amyevans** [comment]
I've updated the expected results in the OP to capture what @roryabraham shared in Slack [here](https://expensify.slack.com/archives/C049HHMV9SM/p1674497498715339?thread_ts=1674469304.602449&cid=C049HHMV9SM).

Since @aimane-chnaif has full context, I'll assign you as C+. And we can open up to external contributors.

**Comment by s77rt** [comment]
### Proposal
#### Part 1
```diff
diff --git a/src/CONST.js b/src/CONST.js
index 0a84a4c108..34d7a5cd02 100755
--- a/src/CONST.js
+++ b/src/CONST.js
@@ -523,6 +523,7 @@ const CONST = {
             3: 100,
         },
     },
+    OPTION_LIST_SKELETON_VIEW_ROW_HEIGHT: 60,
     EXPENSIFY_PARTNER_NAME: 'expensify.com',
     EMAIL: {
         CONCIERGE: 'concierge@expensify.com',
diff --git a/src/components/OptionsList/BaseOptionsList.js b/src/components/OptionsList/BaseOptionsList.js
index b7b6ebcd5a..aa8a60612d 100644
--- a/src/components/OptionsList/BaseOptionsList.js
+++ b/src/components/OptionsList/BaseOptionsList.js
@@ -7,6 +7,7 @@ import variables from '../../styles/variables';
 import OptionRow from '../OptionRow';
 import SectionList from '../SectionList';
 import Text from '../Text';
+import OptionsListSkeletonView from './OptionsListSkeletonView';
 import {propTypes as optionsListPropTypes, defaultProps as optionsListDefaultProps} from './optionsListPropTypes';
 
 const propTypes = {
@@ -42,13 +43,18 @@ class BaseOptionsList extends Component {
         this.didLayout = false;
 
         this.flattenedData = this.buildFlatSectionArray();
+
+        this.state = {
+            skeletonViewContainerHeight: 0,
+        };
     }
 
-    shouldComponentUpdate(nextProps) {
+    shouldComponentUpdate(nextProps, nextState) {
         return nextProps.focusedIndex !== this.props.focusedIndex
             || nextProps.selectedOptions.length !== this.props.selectedOptions.length
             || nextProps.headerMessage !== this.props.headerMessage
-            || !_.isEqual(nextProps.sections, this.props.sections)
+            || nextProps.isLoadingReportData !== this.props.isLoadingReportData
+            || nextState.skeletonViewContainerHeight !== this.state.skeletonViewContainerHeight
             || !_.isEqual(nextProps.sections, this.props.sections);
     }
 
@@ -202,36 +208,54 @@ class BaseOptionsList extends Component {
 
     render() {
         return (
-            <View style={this.props.listContainerStyles}>
-                {this.props.headerMessage ? (
-                    <View style={[styles.ph5, styles.pb5]}>
-                        <Text style={[styles.textLabel, styles.colorMuted]}>
-                            {this.props.headerMessage}
-                        </Text>
-                    </View>
-                ) : null}
-                <SectionList
-                    ref={this.props.innerRef}
-                    indicatorStyle="white"
-                    keyboardShouldPersistTaps="always"
-                    keyboardDismissMode={this.props.keyboardDismissMode}
-                    onScrollBeginDrag={this.props.onScrollBeginDrag}
-                    onScroll={this.props.onScroll}
-                    contentContainerStyle={this.props.contentContainerStyles}
-                    showsVerticalScrollIndicator={false}
-                    sections={this.props.sections}
-                    keyExtractor={this.extractKey}
-                    stickySectionHeadersEnabled={false}
-                    renderItem={this.renderItem}
-                    getItemLayout={this.getItemLayout}
-                    renderSectionHeader={this.renderSectionHeader}
-                    extraData={this.props.focusedIndex}
-                    initialNumToRender={12}
-                    maxToRenderPerBatch={5}
-                    windowSize={5}
-                    viewabilityConfig={{viewAreaCoveragePercentThreshold: 95}}
-                    onViewableItemsChanged={this.onViewableItemsChanged}
-                />
+            <View
+                style={this.props.listContainerStyles}
+                onLayout={(event) => {
+                    const skeletonViewContainerHeight = event.nativeEvent.layout.height;
+
+                    // The height can be 0 if the component unmounts - we are not interested in this value and want to know how much space it
+                    // takes up so we can set the skeleton view container height.
+                    if (skeletonViewContainerHeight === 0) {
+                        return;
+                    }
+                    this.setState({skeletonViewContainerHeight});
+                }}
+            >
+                {this.props.isLoadingReportData ? (
+                    <OptionsListSkeletonView containerHeight={this.state.skeletonViewContainerHeight} />
+                ) : (
+                    <>
+                        {this.props.headerMessage ? (
+                            <View style={[styles.ph5, styles.pb5]}>
+                                <Text style={[styles.textLabel, styles.colorMuted]}>
+                                    {this.props.headerMessage}
+                                </Text>
+                            </View>
+                        ) : null}
+                        <SectionList
+                            ref={this.props.innerRef}
+                            indicatorStyle="white"
+                            keyboardShouldPersistTaps="always"
+                            keyboardDismissMode={this.props.keyboardDismissMode}
+                            onScrollBeginDrag={this.props.onScrollBeginDrag}
+                            onScroll={this.props.onScroll}
+                            contentContainerStyle={this.props.contentContainerStyles}
+                            showsVerticalScrollIndicator={false}
+                            sections={this.props.sections}
+                            keyExtractor={this.extractKey}
+                            stickySectionHeadersEnabled={false}
+                            renderItem={this.renderItem}
+                            getItemLayout={this.getItemLayout}
+                            renderSectionHeader={this.renderSectionHeader}
+                            extraData={this.props.focusedIndex}
+                            initialNumToRender={12}
+                            maxToRenderPerBatch={5}
+                            windowSize={5}
+                            viewabilityConfig={{viewAreaCoveragePercentThreshold: 95}}
+                            onViewableItemsChanged={this.onViewableItemsChanged}
+                        />
+                    </>
+                )}
             </View>
         );
     }
diff --git a/src/components/OptionsList/OptionsListSkeletonView.js b/src/components/OptionsList/OptionsListSkeletonView.js
new file mode 100644
index 0000000000..d316cfc1c6
--- /dev/null
+++ b/src/components/OptionsList/OptionsListSkeletonView.js
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Rect, Circle} from 'react-native-svg';
+import SkeletonViewContentLoader from 'react-content-loader/native';
+import themeColors from '../../styles/themes/default';
+import styles from '../../styles/styles';
+import CONST from '../../CONST';
+
+const propTypes = {
+    /** Height of the container component */
+    containerHeight: PropTypes.number.isRequired,
+
+    /** Whether to animate the skeleton view */
+    animate: PropTypes.bool,
+};
+
+const defaultProps = {
+    animate: true,
+};
+
+const OptionsListSkeletonView = (props) => {
+    // Determines the number of content items based on container height
+    const possibleVisibleContentItems = Math.ceil(props.containerHeight / CONST.OPTION_LIST_SKELETON_VIEW_ROW_HEIGHT);
+
+    const skeletonOptionsList = [];
+    for (let index = 0; index < possibleVisibleContentItems; index++) {
+        skeletonOptionsList.push(
+            <SkeletonViewContentLoader
+                animate={props.animate}
+                height={CONST.OPTION_LIST_SKELETON_VIEW_ROW_HEIGHT}
+                backgroundColor={themeColors.highlightBG}
+                foregroundColor={themeColors.border}
+                style={styles.mr5}
+                key={`skeletonOptionsList${index}`}
+            >
+                <Circle cx="40" cy="26" r="20" />
+                <Rect x="67" y="11" width="20%" height="8" />
+                <Rect x="67" y="31" width="100%" height="8" />
+            </SkeletonViewContentLoader>,
+        );
+    }
+
+    return <>{skeletonOptionsList}</>;
+};
+
+OptionsListSkeletonView.displayName = 'OptionsListSkeletonView';
+OptionsListSkeletonView.propTypes = propTypes;
+OptionsListSkeletonView.defaultProps = defaultProps;
+export default OptionsListSkeletonView;
diff --git a/src/components/OptionsList/index.js b/src/components/OptionsList/index.js
index 31128766f6..5e6d4c2fbf 100644
--- a/src/components/OptionsList/index.js
+++ b/src/components/OptionsList/index.js
@@ -1,8 +1,11 @@
 import React, {Component, forwardRef} from 'react';
 import {Keyboard} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
 import _ from 'underscore';
 import BaseOptionsList from './BaseOptionsList';
 import withWindowDimensions from '../withWindowDimensions';
+import compose from '../../libs/compose';
+import ONYXKEYS from '../../ONYXKEYS';
 import {propTypes, defaultProps} from './optionsListPropTypes';
 import * as DeviceCapabilities from '../../libs/DeviceCapabilities';
 
@@ -65,7 +68,14 @@ OptionsList.propTypes = {
 };
 OptionsList.defaultProps = defaultProps;
 
-export default withWindowDimensions(forwardRef((props, ref) => (
+export default compose(
+    withWindowDimensions,
+    withOnyx({
+        isLoadingReportData: {
+            key: ONYXKEYS.IS_LOADING_REPORT_DATA,
+        },
+    }),
+)(forwardRef((props, ref) => (
     // eslint-disable-next-line react/jsx-props-no-spreading
     <OptionsList forwardedRef={ref} {...props} />
 )));
diff --git a/src/components/OptionsList/index.native.js b/src/components/OptionsList/index.native.js
index 7701096fde..b5326854e2 100644
--- a/src/components/OptionsList/index.native.js
+++ b/src/components/OptionsList/index.native.js
@@ -1,6 +1,8 @@
 import React, {forwardRef} from 'react';
 import {Keyboard} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
 import BaseOptionsList from './BaseOptionsList';
+import ONYXKEYS from '../../ONYXKEYS';
 import {propTypes, defaultProps} from './optionsListPropTypes';
 
 const OptionsList = forwardRef((props, ref) => (
@@ -16,4 +18,8 @@ OptionsList.propTypes = propTypes;
 OptionsList.defaultProps = defaultProps;
 OptionsList.displayName = 'OptionsList';
 
-export default OptionsList;
+export default withOnyx({
+    isLoadingReportData: {
+        key: ONYXKEYS.IS_LOADING_REPORT_DATA,
+    },
+})(OptionsList);
diff --git a/src/components/OptionsList/optionsListPropTypes.js b/src/components/OptionsList/optionsListPropTypes.js
index 0ca8b7a709..a27bfe80f7 100644
--- a/src/components/OptionsList/optionsListPropTypes.js
+++ b/src/components/OptionsList/optionsListPropTypes.js
@@ -70,6 +70,8 @@ const propTypes = {
 
     /** Whether to show a line separating options in list */
     shouldHaveOptionSeparator: PropTypes.bool,
+
+    isLoadingReportData: PropTypes.bool,
 };
 
 const defaultProps = {
@@ -90,6 +92,7 @@ const defaultProps = {
     isDisabled: false,
     onLayout: undefined,
     shouldHaveOptionSeparator: false,
+    isLoadingReportData: false,
 };
 
 export {propTypes, defaultProps};
```

In this part we will handle the core feature request where I decided to implement the skeleton view globally on `<OptionsList />` to cover similar issues on other places.

First I created a new skeleton component `<OptionsListSkeletonView />`
This skeleton view will only be shown if reports are still being loaded and for that we are going to use onyx (key:  `IS_LOADING_REPORT_DATA`)

Basically that's all what part 1 is about. The skeleton height calculation logic is the same as on `<ReportScreen />` where we use `<View />`'s `onLayout` method to get the container height and draw `X` rows where `X = container height / height of one row`


#### Part 2
```diff
diff --git a/src/pages/SearchPage.js b/src/pages/SearchPage.js
index 3cefc6a98c..3c0965c4a1 100755
--- a/src/pages/SearchPage.js
+++ b/src/pages/SearchPage.js
@@ -30,6 +30,8 @@ const propTypes = {
     /** All of the personal details for everyone */
     personalDetails: personalDetailsPropType.isRequired,
 
+    isLoadingReportData: PropTypes.bool,
+
     /** All reports shared with the user */
     reports: PropTypes.objectOf(reportPropTypes).isRequired,
 
@@ -44,6 +46,10 @@ const propTypes = {
     ...withLocalizePropTypes,
 };
 
+const defaultProps = {
+    isLoadingReportData: false,
+};
+
 class SearchPage extends Component {
     constructor(props) {
         super(props);
@@ -75,6 +81,13 @@ class SearchPage extends Component {
         };
     }
 
+    componentDidUpdate(prevProps) {
+        if (prevProps.isLoadingReportData === this.props.isLoadingReportData) {
+            return;
+        }
+        this.debouncedUpdateOptions();
+    }
+
     onChangeText(searchValue = '') {
         this.setState({searchValue}, this.debouncedUpdateOptions);
     }
@@ -198,11 +211,15 @@ class SearchPage extends Component {
 }
 
 SearchPage.propTypes = propTypes;
+SearchPage.defaultProps = defaultProps;
 
 export default compose(
     withLocalize,
     withWindowDimensions,
     withOnyx({
+        isLoadingReportData: {
+            key: ONYXKEYS.IS_LOADING_REPORT_DATA,
+        },
         reports: {
             key: ONYXKEYS.COLLECTION.REPORT,
         },
```
This part is specifically for `<SearchPage />` as this page suffers from stale data. Meaning if reports gets loaded/updated after the search page is already rendered, it won't re-render to match the new reports and it will be out of sync.

As @mallenexpensify  explained we have a workaround here where if we type in the search input you will get the expected results that's because changing the input will call `debouncedUpdateOptions` which rebuilds the list using fresh data. Fixing the stale data issue can be done in many ways as all we have to do is to call `debouncedUpdateOptions`.

Since we are only looking to rebuild the list if reports load after render, we can use the same onyx key we used before (`IS_LOADING_REPORT_DATA`).

I think the code in this part is self-explanatory.

#### Results

https://user-images.githubusercontent.com/16493223/215767770-babf8b9c-2d57-4ec4-9f9b-0f2a34206d86.mp4

**Comment by amyevans** [comment]
Understanding that implementation here will ultimately be on hold for the merge of https://github.com/Expensify/App/pull/14456, but @aimane-chnaif can you review the above proposal in the meantime?

**Comment by MelvinBot** [comment]
@amyevans @mallenexpensify @aimane-chnaif this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
@amyevans @mallenexpensify @aimane-chnaif this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by mallenexpensify** [comment]
I added `[HOLD #14456]` to the title so the issue won't double in price

**Comment by amyevans** [comment]
Brought the question to Slack [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1675807968905499) for a gut check.

**Comment by amyevans** [comment]
Not a lot of feedback on that question, but I think it'd make sense to proceed as follows:
* Once https://github.com/Expensify/App/pull/14456 is merged, create a separate cleanup issue to evaluate all pages for stale data
* Hold this issue on ^, so that the scope of this issue does not balloon beyond its initial bug report

Moving this to weekly in the interim.

**Comment by amyevans** [comment]
Not overdue, still on hold.

**Comment by amyevans** [comment]
On hold

**Comment by mallenexpensify** [comment]
The PR that we're held on is making progress, was reviewed yesterday https://github.com/Expensify/App/pull/14456

**Comment by amyevans** [comment]
Still holding

**Comment by mallenexpensify** [comment]
ooooh, https://github.com/Expensify/App/pull/14456 hit production two days ago. Should we take this off hold now?!?!

**Comment by amyevans** [comment]
FYI I'll be OOO next week - feel free to reassign as needed @mallenexpensify to keep things moving

**Comment by priyeshshah11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Opening any shortcut pages straight after login doesn't show any data.

### What is the root cause of that problem?

The issue is that it takes some time for all the data to be loaded before it's displayed.

### What changes do you think we should make in order to solve the problem?

We should display a loading spinner/activity indicator in those pages/sidebars/lists until the data has been fetched from the backend & is ready to be displayed.

**Comment by mountiny** [comment]
Assigning myself to help with this issue when Amy is ooo

**Comment by mczernek** [comment]
Hi, I am Michal from Software Mansion - an expert contributor - and I will investigate this issue.

**Comment by MelvinBot** [comment]
📣 @mczernek You have been assigned to this job by @mountiny!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by bernhardoj** [comment]
@aimane-chnaif I think we can exclude the IOU page because we have this issue https://github.com/Expensify/App/issues/16739

**Comment by bernhardoj** [comment]
Ah you're right, it use the same component, we just can't test the IOU page here.

**Comment by bernhardoj** [comment]
My bad, I misunderstood your statement. Yes, the loading issue will be fixed but the crash won't just get away as I can say it's not "related" to the empty list. Should I post my finding here https://github.com/Expensify/App/issues/16739 for better context?

**Comment by situchan** [comment]
Is this open for proposals? I have a full proposal to submit.

**Comment by mountiny** [comment]
@situchan sorry, we exported this issue to SWM because it was breaking WAQ since its over 5 weeks old at this point so we will have this dedicated person see the end of this issue. I hope your proposal will be accepted on the other issue!

**Comment by jczekalski** [comment]
Hey @mountiny, can you reassign the issue to me?

**Comment by MelvinBot** [comment]
📣 @jczekalski! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by jczekalski** [comment]
**Update**: I have a general idea how to fix the issue and I'm working on a proposal.

So far I've reproduced the issue on these pages:
- SearchPage
- NewChatPage, NewGroupPage
- IOURequestPage, IOUSendPage

I couldn't reproduce it on the Profile and Workspaces pages, but perhaps that's because the Settings page takes a while to load before those pages can be accessed. I also noticed the Settings screen doesn't have a skeleton loading state (video); but that feels like a separate issue?

https://user-images.githubusercontent.com/39824924/230645640-f6ed0318-0f0e-4832-92d0-eb58e2321760.mov

**Comment by MelvinBot** [comment]
📣 @jczekalski You have been assigned to this job by @mountiny!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by amyevans** [comment]
@jczekalski The proposal looks good to me, please move forward with opening a PR if you haven't already. @aimane-chnaif please still review the proposal and provide feedback if you have any. Just a reminder this issue is technically breaking WAQ so let's collectively focus on closing it as a high priority 😄 . Thanks guys! 🙏

**Comment by jczekalski** [comment]
> Looks good to me in general. @jczekalski no need full code diff in proposal stage as stated in guideline. Here's my concern:
> 
> * If login same account on another device and adds more contacts, list isn't synced with latest data until user types any letter. If we do this change, I believe it will cause regression.
> 
> ```diff
> -            || !_.isEqual(nextProps.sections, this.props.sections)
> +            || nextProps.isLoading !== this.props.isLoading
> ```
> 
> * We should avoid calling `getNewChatOptions`, `getSearchOptions` which requires reports and personalDetails before getting them after login
> * I think some other pages like Settings should also be covered here
> 
> Anyway, we can tackle them in PR. Let's go ahead 🚀

@aimane-chnaif Regarding your first comment, my suggestion was not to remove this check
```
!_.isEqual(nextProps.sections, this.props.sections)
```

This condition is actually checked twice in `shouldComponentUpdate`, so I replaced the duplicate with `nextProps.isLoading !== this.props.isLoading`.

Thank you for reviewing the proposal, I'll open a PR and we can discuss this everything in more detail 👍

**Comment by MelvinBot** [comment]
@amyevans, @mallenexpensify, @jczekalski, @aimane-chnaif Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by amyevans** [comment]
PR is being actively worked on.

**Comment by MelvinBot** [comment]
@amyevans, @mallenexpensify, @jczekalski, @aimane-chnaif 8 days overdue is a lot. Should this be a Weekly issue? If so, feel free to change it!

**Comment by amyevans** [comment]
PR is in review, I still personally need to get to it today though.

**Comment by kavimuru** [comment]
Keyboard short cut bugs are not in scope. Closing the bug as per this [comment](https://expensify.slack.com/archives/C049HHMV9SM/p1683574980798919?thread_ts=1683482157.624799&cid=C049HHMV9SM)

**Comment by mountiny** [comment]
The regression here was impossible to catch so the payments should not be degraded because of it

**Comment by amyevans** [comment]
Bumped @mallenexpensify in New Dot for payment

**Comment by mallenexpensify** [comment]
Thanks @amyevans , @aimane-chnaif, can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~0193e6015e52379df0

@aimane-chnaif can you address the items in the BZ checklist above too please?
https://github.com/Expensify/App/issues/14490#issuecomment-1547376158

**Comment by mallenexpensify** [comment]
Thanks @aimane-chnaif , TC GH is here
- https://github.com/Expensify/Expensify/issues/286289 

Let me know when you've accepted in Upwork and I'll pay

**Comment by mallenexpensify** [comment]
@aimane-chnaif paid $4000. I wasn't 100% sure if this is accurate but, it seemed _fair_ upon a cursory review as the associated PR fixed a few related issues. My thinking was that any negative timeliness bonuses would be negated due to the PR fixing multiple issues.  If anyone disagrees, please comment.

**Comment by mallenexpensify** [comment]
>And this is minor but I might also be eligible for reporting bonus.

@aimane-chnaif , you def were/are.  I missed it, apologies.  Just paid ya $250.
